CRM/Contribute/BAO/Contribution.php

```diff
@@ -183,12 +183,17 @@ public static function add(&$params, $ids = array()) {
       }
     }
     if ($contributionID && $setPrevContribution) {
-      $params['prevContribution'] = self::getValues(array('id' => $contributionID), CRM_Core_DAO::$_nullArray, CRM_Core_DAO::$_nullArray);
+      $params['prevContribution'] = self::getOriginalContribution($contributionID);
     }
 
     // CRM-16189
     CRM_Financial_BAO_FinancialAccount::checkFinancialTypeHasDeferred($params, $contributionID);
 
+    if (!isset($params['tax_amount']) && $setPrevContribution && (isset($params['total_amount']) || isset
+      ($params['financial_type_id']))) {
+      $params = CRM_Contribute_BAO_Contribution::checkTaxAmount($params);
```

Comment:
@eileenmcnaughton i was finding that the $params being passed to the copyValues was the original and not updated to include the relevant tax amount for some reason